### DISCUSSION TITLE
DCHP reservations during install

### DISCUSSION THREAD

**Discussion by AlexGeek77** [tteck/Proxmox#82] [Ideas]
I typically create DHCP reservations for each of my LXCs but I can only do that after the install scripts complete. That then requires me to release the DHCP reservation in the container. It's not really too much of an issue though I did notice in the Wireguard install I found the WGDashboard grabbed the current IP address and coded it in to the Peer Remote Endpoint variable in the settings. Most install scripts show the IPv4 and IPv6 addresses received. It would be nice to show the MAC address and then prompts the user to press a key to continue. That would give the user time to create the DHCP reservation prior to the install script starting up the container.

**Comment by tteck** [comment]
~The best that I'll do is show the MAC address with the IP address.~
WireGuard LXC script is setup that way now if you'd like to test it out
~I doubt that helps, but...~

**Comment by tteck** [comment]
@zelig2 I fixed the script to do exactly what you asked. All that I need is for someone to test and let me know if it's working as intended.

![Screenshot 2022-03-20 12 06 23 PM](https://user-images.githubusercontent.com/21342670/159171509-fee6569d-4fa1-45df-a71e-e8ab66e48bde.png)

**Reply by mschabhuettl** [reply]
Wow! Thank you very much! Actually I need this feature, so in the past i just manually added a "wait"-sequence in your script, to manually do that.

**Reply by AlexGeek77** [reply]
Agreed, it works great for me as well. Thanks for making this change. I've reinstalled Dashy and installed Vaultwarden and it made the processes more streamlined.

**Comment by tteck** [comment]
FWIW, if you select "Manual DHCP" during install and when it asks "Please set DHCP reservation and press Enter." you can open a new tab, open Proxmox and edit that LXC (Bridge, VLAN Tag, hostname whatever) then go back to the original Proxmox tab and press enter to start the LXC with your changed settings.

DO NOT leave the shell that you ran the script in. If you do you'll have to remove that CT and start over.
